### PR TITLE
added UrlConventions support in DefaultODataPathHandler

### PR DESCRIPTION
**Pull request by BarclayII**

### CODE REVIEW

File: OData/src/System.Web.OData/OData/Routing/DefaultODataPathHandler.cs [hunk 1]
```diff
@@ -31,6 +32,15 @@ internal ODataUriResolverSetttings ResolverSetttings
         }
 
         /// <summary>
+        /// Gets or sets the <see cref="ODataUrlConventions"/> used by the parser.
+        /// </summary>
+        public ODataUrlConventions UrlConventions
```
Comment (congysu) [inline OData/src/System.Web.OData/OData/Routing/DefaultODataPathHandler.cs]: Enable key as segment, while the OData () key syntax still works.

**Comment by congysu** [comment]
Pending the new ODataSimplified key-as-segment convention to be released in OData v4 Lib 6.14.

**Comment by congysu** [comment]
OData v4 Lib 6.14 released.  For new key-as-segment convention, please use [ODataUrlConventions ODataSimplified](https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/UriParser/ODataUrlConventions.cs#L58).

**Comment by VikingsFan** [comment]
Merged 33c370b79fe055a57fbd6e3767a007437f2f8c41

**Pull request closed without merging**